compiler/src/iree/compiler/GlobalOptimization/DecomposeLowParallelismAttentionPass.cpp

```diff
@@ -0,0 +1,118 @@
+// Copyright 2024 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "iree/compiler/Dialect/LinalgExt/IR/LinalgExtOps.h"
+#include "iree/compiler/Dialect/LinalgExt/Utils/IndexingUtils.h"
+#include "iree/compiler/GlobalOptimization/Passes.h"
+
+namespace mlir::iree_compiler::GlobalOptimization {
+
+#define GEN_PASS_DEF_DECOMPOSELOWPARALLELISMATTENTIONPASS
+#include "iree/compiler/GlobalOptimization/Passes.h.inc"
+
+using namespace IREE::LinalgExt;
+
+namespace {
+
+const int64_t kDynamicDimMaxValue = 65536; // 2^16
+/// TODO: The M heuristic is a conservative choice I(Groverkss) made.
+/// We ideally need more data to determine when it is good to
+/// decompose.
+const int64_t kMParallelismThreshold = 4;
```

Comment:
I think this could be high as 16 (least m for cooperative subgroup matmul) x 4 (number of waves) = 64.
Do we have compilation path that does not use cooperative subgroup matmul for fused attention ?

Reply:
I kind of like 4 since it matches with https://github.com/iree-org/iree/blob/0077358221e3fd2a52d114115ac9b9d17089fc16/compiler/src/iree/compiler/Codegen/LLVMGPU/KernelConfig.cpp#L93-L94

Reply:
IIUC, that is used to send skinny matmuls in a vector reduction pipeline where is there is only one pipeline for attention : VectorDistribute. 
Another way to rephrase `Do we have compilation path that does not use cooperative subgroup matmul for fused attention ?`, is to ask whether VectorDistribute can lower w/o MFMA configs set on it ?